test/extended/README.md

```diff
@@ -0,0 +1,145 @@
+* [**`hack/test-extended/[group]/run.sh`**](../../hack/test-extended) contains a shell script that helps you to execute extended tests for a certain group
+* [**`test/extended/extended_test.go`**](extended_test.go) is a runner for all extended test packages. Looking inside this file, you can see how you can add new extended test Go package to be compiled:
+```go
+	_ "github.com/openshift/origin/test/extended/builds"
+	_ "github.com/openshift/origin/test/extended/images"
+```
+
+Groups vs. packages
+---------------------
+
+Since the extended tests might rely on specify OpenShift server configuration,
+the tests are divided into logical 'test groups'. Each group has its own shell
+launcher that bootstraps the OpenShift environment in a way the group requires
+to be executed.
+For example, you might want to write an extended test for the LDAP
+authentication which means that you have to configure OpenShift server to
+enable this authentication method. 
+You can create a new test group `ldap` and provide a shell launcher
+`./hack/test-extended/ldap/run.sh` to start OpenShift with required
+configuration.
+Then you place the source code with extended test into the extended test Go
+package that corresponds to functionality you are going to test. In case of
+LDAP, it can be `./test/extended/authentication`.  In order to have your test
+case executed by `./hack/test-extended/ldap/run.sh` you have to add `ldap:`
+prefix to the `Describe()` function:
+
+Example:
+```go
+var _ = g.Describe("ldap: Authenticate using LDAP", func() {
+  # ...
+})
+```
+
+
+CLI interface
+---------------------
+
+In order to be able to call the OpenShift CLI and Kubernetes and OpenShift REST client and simulate the OpenShift `oc` command in the test suit, first we need to create an instance of the CLI, in the top-level Ginkgo describe container. The top-level describe container shall also specify the the bucket, into which the test belongs and short test description. Other globally accessible variables can be declared(eg. fixtures) can be declared as well.
+
+```go
+package extended
+
+import (
+    g "github.com/onsi/ginkgo"
+    o "github.com/onsi/gomega"
+)
+
+var _ = g.Describe("<test bucket>: <Testing scenario>", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc = exutil.NewCLI("test-name", exutil.KubeConfigPath())
+		testFixture = filepath.Join("fixtures", "test.json")
+	)
+})
+```
+
+The test suit shall be organized in lower-level Ginkgo describe container, together with an message which informs about the goal of the test. Inside the lower-level describe container specify a single spec with the `It` container , which shares the context in which the spec runs. The `It` container also takes a message, which informs how should be the goal achieved.
```

Comment:
s/ test suit shall / test suite should /
s/ organized in / organized into /
s/ describe container / describe container(s) /
s/ an message / a message /
s/ informs about / elaborates on /
s/ Inside the / Inside each /
s/ takes a message, which informs how should be the goal achieved / takes a message which explains how the test goal will be achieved /